internal/helm/controller/reconcile.go

```diff
@@ -247,7 +247,7 @@ func (r HelmOperatorReconciler) Reconcile(request reconcile.Request) (reconcile.
 			log.Error(err, "Release failed")
 			status.SetCondition(types.HelmAppCondition{
 				Type:    types.ConditionReleaseFailed,
-				Status:  types.StatusTrue,
+				Status:  types.StatusFalse,
```

Comment:
This doesn't align with your PR description.

The PR description talks about the `Initialized` Condition type, but this is changing the `ReleaseFailed` condition.

If the upgrade fails, wouldn't you expect `ReleaseFailed` to be `True`?